lib/aposPageFetch.js

```diff
@@ -6,17 +6,22 @@ export default async function aposPageFetch(req) {
   try {
     const request = aposRequest(req);
     const response = await aposResponse(request);
+    if (!response.ok) {
+      const errorMessage = await response.text();
+      throw new Error(errorMessage);
+    }
```

Comment:
that's important otherwise we get an error on the line below where we try to jsonify something that's not a json!